### ISSUE TITLE
Querying attributes from a Bullet manifold which is stored in a Dict that is stored in a List crashes panda3D

### ISSUE THREAD

**Issue by K9Kraken** [panda3d/panda3d#1192]
## Description
Discussion on this issue [here](https://discourse.panda3d.org/t/confusing-strange-crash-with-bullet-manifold-along-with-python-list-and-dict/28073/7)

Panda3D crashing when all this is True:
- Creating two Bullet objects with rigid bodies made from a Triangle mesh.
- Do a contact test on the objects then store the contact manifold in a DICT that is stored in a LIST
- Then trying to access a attribute from that manifold will crash Panda3D.

Making a collider egg file and using BulletHelper.fromCollisionSolids to make a rigid body also results in a crash.

If you make the bodies from convex hulls it works fine.

If you don't store the contact manifold class inside a DICT that is also stored in a LIST then accessing it's attributes works fine.

Its as if for some reason storing the contact manifold class inside a DICT and LIST somehow messes up pointers within the class.

Tested same version of python and Panda3D on a Windows machine and it worked without issue.

## Steps to Reproduce
```
ffrom direct.showbase.ShowBase import ShowBase
from panda3d.core import NodePath
from panda3d.bullet import BulletWorld
from panda3d.bullet import BulletBoxShape, BulletRigidBodyNode, BulletConvexHullShape, BulletTriangleMeshShape, BulletTriangleMesh


class Game(ShowBase):
    def __init__(self):
        ShowBase.__init__(self)
        base.disableMouse()
        camera.setPos(0,-20,0)
        world = BulletWorld()

        self.world = world

        self.node = NodePath('ph')
        self.node.reparent_to(render)

        self.b1 = self.make_ball(-2, 0, 0)
        self.b2 = self.make_ball(2, 0, 0)
        self.b2.node().set_kinematic(True)
        self.b1.node().set_kinematic(True)

        taskMgr.add(self.update, 'update')




    def make_ball(self, x, y, z):
        smiley = loader.loadModel("smiley")

        geomNodes = smiley.findAllMatches('**/+GeomNode')
        geomNode = geomNodes.getPath(0).node()
        geom = geomNode.getGeom(0)

        # Make Convex Shape:
        #shape = BulletConvexHullShape()
        #shape.addGeom(geom)

        # Make TriangleMesh Shape:
        shape = BulletTriangleMesh()
        shape.add_geom(geom)
        shape = BulletTriangleMeshShape(shape, dynamic=True)


        body = BulletRigidBodyNode()
        body.add_shape(shape)

        body.set_mass(1.0)

        self.ball = NodePath(body)

        smiley.reparentTo(self.ball)
        self.ball.set_pos( (x, y, z) )

        self.world.attach_rigid_body(body)
        self.ball.reparent_to(self.node)

        return self.ball


    def update(self, task):
        self.world.do_physics(globalClock.get_dt())


        result = self.world.contact_test(self.b1.node())
        for contact in result.get_contacts():
            mpoint = contact.get_manifold_point()
            print("A1")
            print(mpoint.distance)
            print("A2")

        hits = []
        h = None
        for contact in result.get_contacts():
            hit = {}
            hit['MPOINT'] = contact.manifold_point
            print("B1")
            print(hit['MPOINT'].distance)
            print("B2")
            hits.append(hit)
            h = hit

        if h:
            print("C1")
            print(h['MPOINT'].distance)
            print("C2")

        if hits:
            for hit in hits:
                print("E1")
                # Crash happens here if using triangle hull:
                print(hit['MPOINT'].distance)
                print("E2")





        pos = self.b1.get_pos()
        pos.x += 0.01
        self.b1.set_pos(pos)

        #self.b1.node().set_linear_velocity((1,0,0))
        return task.cont




game = Game()
game.run()
```

## Environment
 * Operating system: Manjaro
 * System architecture: x86_64
 * Panda3D version: 1.10.10
 * Installation method: pip
 * Python version: 3.9.7

**Comment by Moguri** [comment]
@rdb, do we close this as "cannot reproduce"? Maybe making the change anyways to fix other potential issues?

**Comment by rdb** [comment]
Possibly addressed by f64dcbd3df9246d2cc4f39e56f9986b3d3f453de, possibly not.  It's recommended to keep the contact around anyway rather than storing the manifold point.  Closing, we can reopen if we get a reproducible test case.
